src/runtime/dynlink/src/tls.rs

```diff
@@ -171,6 +172,13 @@ impl<T> Tcb<T> {
             runtime_data: tcb_data,
         }
     }
+
+    pub fn get_addr(&self, index: &TlsIndex) -> *const u8 {
+        unsafe {
+            let slice = core::slice::from_raw_parts(self.dtv, self.dtv_len);
+            (slice[index.mod_id] + index.offset) as *const _
```

Comment:
Should we consider using `get` here? Is `mod_id` guaranteed to be within the bounds/less than `dtv_len`?

Reply:
Nice catch, in principal it's supposed to be, but malicious code could trigger a panic. Will fix.